tidy3d/components/parameter_perturbation.py

```diff
@@ -1227,6 +1228,389 @@ def _sample_delta_eps_delta_sigma(
+                    0.985,
+                    0.973,
+                    0.968,
+                    0.965,
+                    0.964,
+                    0.969,
+                    0.984,
+                    0.984,
+                    0.966,
+                    0.948,
+                    0.951,
+                    0.944,
+                    0.934,
+                    0.965,
+                    0.958,
+                    0.953,
+                    0.945,
+                    0.936,
+                    0.928,
+                    0.922,
+                ],
+            ),
+            "r": (
+                "wvl",
+                [
+                    1.25e-18,
+                    1.53e-18,
+                    2.28e-18,
+                    5.19e-18,
+                    3.62e-18,
+                    9.28e-18,
+                    9.99e-18,
+                    1.29e-17,
+                    2.03e-17,
+                    3.31e-17,
+                    6.92e-17,
+                    8.23e-17,
+                    1.15e-16,
+                    4.81e-16,
+                    7.44e-16,
+                    7.11e-16,
+                    5.29e-16,
+                    9.72e-16,
+                    1.22e-15,
+                    1.16e-15,
+                    3.16e-15,
+                    1.51e-14,
+                    2.71e-14,
+                    2.65e-14,
+                    2.94e-14,
+                    6.85e-14,
+                    2.60e-13,
+                ],
+            ),
+            "s": (
+                "wvl",
+                [
+                    0.835,
+                    0.838,
+                    0.841,
+                    0.832,
+                    0.849,
+                    0.834,
+                    0.839,
+                    0.838,
+                    0.833,
+                    0.826,
+                    0.812,
+                    0.812,
+                    0.807,
+                    0.776,
+                    0.769,
+                    0.774,
+                    0.783,
+                    0.772,
+                    0.769,
+                    0.772,
+                    0.750,
+                    0.716,
+                    0.704,
+                    0.706,
+                    0.705,
+                    0.686,
+                    0.656,
+                ],
+            ),
+        },
+        coords={"wvl": np.array([1.3, 1.55] + list(np.arange(2, 14.5, 0.5)))},
+    )
+
+    ref_freq: pd.NonNegativeFloat = pd.Field(
+        ...,
+        title="Reference Frequency",
+        description="Reference frequency to evaluate perturbation at (Hz).",
+        units=HERTZ,
+    )
+
+    @pd.root_validator(skip_on_failure=True)
+    def _check_freq_in_range(cls, values):
+        """Check that the given frequency is within validity range.
+        If not, issue a warning.
+        """
+
+        freq = values.get("ref_freq")
+        freq_range = (C_0 / 14, C_0 / 1.3)
```

Comment:
probably better to use `np.min(coords.wvl)` and `np.max(coords.wvl)` so that's it's clear where they are coming from